main.tf

```diff
@@ -3067,3 +3067,118 @@ module "vpa" {
+  create_namespace = try(var.aws_gateway_api_controller.create_namespace, true)
+  chart            = "aws-gateway-controller-chart"
+  chart_version    = try(var.aws_gateway_api_controller.chart_version, "v0.0.12")
+  repository       = try(var.aws_gateway_api_controller.repository, "oci://public.ecr.aws/aws-application-networking-k8s")
+  values           = try(var.aws_gateway_api_controller.values, [])
+
+  timeout                    = try(var.aws_gateway_api_controller.timeout, null)
+  repository_key_file        = try(var.aws_gateway_api_controller.repository_key_file, null)
+  repository_cert_file       = try(var.aws_gateway_api_controller.repository_cert_file, null)
+  repository_ca_file         = try(var.aws_gateway_api_controller.repository_ca_file, null)
+  repository_username        = try(var.aws_gateway_api_controller.repository_username, null)
+  repository_password        = try(var.aws_gateway_api_controller.repository_password, null)
+  devel                      = try(var.aws_gateway_api_controller.devel, null)
+  verify                     = try(var.aws_gateway_api_controller.verify, null)
+  keyring                    = try(var.aws_gateway_api_controller.keyring, null)
+  disable_webhooks           = try(var.aws_gateway_api_controller.disable_webhooks, null)
+  reuse_values               = try(var.aws_gateway_api_controller.reuse_values, null)
+  reset_values               = try(var.aws_gateway_api_controller.reset_values, null)
+  force_update               = try(var.aws_gateway_api_controller.force_update, null)
+  recreate_pods              = try(var.aws_gateway_api_controller.recreate_pods, null)
+  cleanup_on_fail            = try(var.aws_gateway_api_controller.cleanup_on_fail, null)
+  max_history                = try(var.aws_gateway_api_controller.max_history, null)
+  atomic                     = try(var.aws_gateway_api_controller.atomic, null)
+  skip_crds                  = try(var.aws_gateway_api_controller.skip_crds, null)
+  render_subchart_notes      = try(var.aws_gateway_api_controller.render_subchart_notes, null)
+  disable_openapi_validation = try(var.aws_gateway_api_controller.disable_openapi_validation, null)
+  wait                       = try(var.aws_gateway_api_controller.wait, false)
+  wait_for_jobs              = try(var.aws_gateway_api_controller.wait_for_jobs, null)
+  dependency_update          = try(var.aws_gateway_api_controller.dependency_update, null)
+  replace                    = try(var.aws_gateway_api_controller.replace, null)
+  lint                       = try(var.aws_gateway_api_controller.lint, null)
+
+  postrender = try(var.aws_gateway_api_controller.postrender, [])
+  set = concat([
+    {
+      name  = "serviceAccount.name"
+      value = local.aws_gateway_api_controller_service_account
+      }, {
+      name  = "awsRegion"
+      value = local.region
+      }, {
+      name  = "clusterVpcId"
+      value = var.vpc_id
```

Comment:
This var can be empty?

Reply:
Its mandatory on the module boundary. We can make it optional but I am not sure if its worth it since it should already be available in the consumer module. And other add-ons including helm releases may need it.